api/v1alpha1/bundleinstance_types.go

```diff
@@ -46,25 +47,37 @@ const (
 type BundleInstanceSpec struct {
 	// ProvisionerClassName sets the name of the provisioner that should reconcile this BundleInstance.
 	ProvisionerClassName string `json:"provisionerClassName"`
+	// Label selector for Bundles resources
+	Selector *metav1.LabelSelector `json:"selector"`
+	// Template describes the generated Bundle that this instance will manage.
+	Template *BundleTemplate `json:"template"`
+}
 
-	// BundleName is the name of the bundle that this instance is managing on the cluster.
-	BundleName string `json:"bundleName"`
+// BundleTemplate defines the desired state of a Bundle resource
+type BundleTemplate struct {
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	// +kubebuilder:pruning:PreserveUnknownFields
```

Comment:
TODO: move this PreserveUnknownFields kubebuilder marker to the parent BundleTemplate field.